source/Xunit.Gherkin.Quick/ScenarioXunitHook/ScenarioOutline/ScenarioOutlineDiscoverer.cs

```diff
@@ -20,30 +20,36 @@ public ScenarioOutlineDiscoverer(IMessageSink messageSink)
             ITestMethod testMethod,
             IAttributeInfo factAttribute)
         {
-            var feature = new FeatureDiscoveryModel(new FeatureFileRepository("*.feature")).Discover(testMethod.TestClass.Class.ToRuntimeType());
+            var featureClass = testMethod.TestClass.Class.ToRuntimeType();
+            var featureClassInfo = FeatureClassInfo.FromFeatureClassType(featureClass);
 
-            foreach (var scenarioOutline in feature.Children.OfType<ScenarioOutline>())
+            var features = new FeatureDiscoveryModel(new FeatureFileRepository(featureClassInfo.FileNameSearchPattern)).Discover(featureClass);
```

Comment:
Same here. Tight coupling and also code duplication, which proves the point. If you move the logic inside, you will not need to do the same thing in two places.